lib/spack/spack/environment.py

```diff
@@ -36,6 +36,7 @@
 from spack.spec import Spec
 from spack.spec_list import SpecList, InvalidSpecConstraintError
 from spack.variant import UnknownVariantError
+from spack.util.lock import Lock, WriteTransaction
```

Comment:
Yes. Although I don't know how this avoids circular imports: generally any import of the module resolves all of the imports in the module (and I assume that would include even imports that are located inside of class definitions, the exception being imports that are inside method bodies). I think that this does improve the readability in any case (by qualifying the references).

I chose `slock` because I thought that was a bit clearer than `lk`.

Reply:
we use `lk` elsewhere in the code

Reply:
Changed `slock` to `lk`.